router-tests/structured_logging_test.go

```diff
@@ -2237,6 +2237,172 @@ func TestFlakyAccessLogs(t *testing.T) {
 			})
 		})
 
+		t.Run("validate the evaluation of the body expression", func(t *testing.T) {
```

Comment:
Please add a test case that reflect that actual customer use case. We wanted to conditionally log the body in case of an error.